src/wallet/wallet.cpp

```diff
@@ -1660,6 +1660,10 @@ mapSaplingNoteData_t CWallet::FindMySaplingNotes(const CTransaction &tx) const
             if (!result) {
                 continue;
             }
+            auto address = ivk.address(result.get().d);
+            if (address) {
+                CBasicKeyStore::AddSaplingIncomingViewingKey(ivk, address.get());
```

Comment:
Currently this function will add even if the address is already added, it just overwrites, but maybe the `AddSaplingIncomingViewingKey` function should document that this behavior is permitted.

Reply:
Added a comment, and fixed another. Force pushed this change.

Reply:
Remove `CBasicKeyStore::` so that when we add wallet serialization, this call will result in the `wallet.dat` being updated.